src/main/java/com/grookage/fsm/core/engine/StateEngine.java

```diff
@@ -149,6 +149,9 @@ public void fire(final E event, final C context) {
       handleTakeOff(to, context);
       handleStateTransition(event, from, context);
       handleLanding(from, context);
+      if (stateManagementService.getEndStates().contains(to)) {
```

Comment:
Related to the below comment, is this required, wouldn't transitions just do the job?

Reply:
Related to the reply above, I feel, if there is a way to register final state event hanlder specifically, state engine should invoke it too